### PR TITLE
Reproduce #334 in a test case

### PR DESCRIPTION
**Pull request by sodre**
This PR should reproduce issue #334 without having to rely on an s3 connection to a large 19GB file.

### CODE REVIEW

File: fsspec/tests/test_api.py [hunk 1]
```diff
@@ -158,6 +161,57 @@ def test_chained_fs():
...
+        # filename, content
+        z.writestr("foo.txt", "foo.txt")
+        z.writestr("bar.txt", "bar.txt")
+
+    with zipfile.ZipFile(f2, mode="w") as z:
+        with open(f1, "rb") as f:
+            z.writestr("f1.zip", f.read())
+
+    # We expected this to be the correct syntax
+    of = fsspec.open_files(f"zip://*.txt::zip://f1.zip::file://{f2}")
```
Comment (sodre) [inline fsspec/tests/test_api.py:207]: This example works without any issues. This indicates to me that the issue is in the blockcache code base and not in the url parsing/unchaining.

Comment (martindurant) [reply fsspec/tests/test_api.py:207]: In this example you have some txt files in f1.zip, which in turn is within f2.zip? That is how I read that URL.

Comment (sodre) [reply fsspec/tests/test_api.py:207]: Yes, that is correct.

File: fsspec/tests/test_api.py [hunk 2]
```diff
@@ -158,6 +161,57 @@ def test_chained_fs():
...
+
+    assert len(of) == 2
+    for open_file in of:
+        with open_file as f:
+            assert f.read().decode("utf-8") == f.name
+
+
```
Comment (sodre) [inline fsspec/tests/test_api.py:214]:
```suggestion

```

File: fsspec/tests/test_api.py [hunk 3]
```diff
@@ -158,6 +161,56 @@ def test_chained_fs():
...
+            z.writestr("f1.zip", f.read())
+
+    # We expected this to be the correct syntax
+    of = fsspec.open_files(f"zip://*.txt::zip://f1.zip::file://{f2}")
+
+    assert len(of) == 2
+    for open_file in of:
+        with open_file as f:
+            assert f.read().decode("utf-8") == f.name
+
```
Comment (sodre) [inline fsspec/tests/test_api.py:213]:
```suggestion

```

**Comment by martindurant** [comment]
OK with you, @sodre ?

**Comment by sodre** [comment]
Hi @martindurant, it looks like it works now. I am not necessarily proud of the solution, but it is as much as I can do at this point.

**Pull request merged**